### PR TITLE
Move interruption callback to a new file

### PR DESCRIPTION
**Pull request by cua-cua**
New files detections.c and detections.h to manage phototransistor sensors and get distance measurement.

### CODE REVIEW

File: src/detections.c [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#include "detections.h"
+#include <libopencm3/stm32/gpio.h>
+#include <libopencm3/stm32/timer.h>
+/**
+ * @brief TIM1 interruption routine.
```
Comment (Peque) [inline src/detections.c]: Maybe a space (or 2) between the includes and the docstring?

File: src/Makefile [hunk 2]
```diff
@@ -1,3 +1,5 @@
+OBJS = detections.o
```
Comment (Peque) [inline src/Makefile]: I would rather call it "detection" rather than "detections". What you think?

File: src/detections.c (hunk 1 above)
Comment (cua-cua) [reply src/detections.c]: The clang-format formatting changes from 2 spaces to 1,but it is not working from no spaces to one.

File: src/Makefile (hunk 2 above)
Comment (cua-cua) [reply src/Makefile]: Ok, changed (to build the project first a make clean should be done to delete old .o)

File: src/detection.h [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#ifndef __DETECTION_H
+#define __DETECTION_H
+
+#endif /* __DETECTION_H */
```
Comment (Peque) [inline src/detection.h]: Should not we include here the `tim1_up_isr` header? In example:

```c
#ifndef __DETECTION_H
#define __DETECTION_H


void tim1_up_isr(void);


#endif /* __DETECTION_H */
```

Comment (cua-cua) [reply src/detection.h]: Done :)

**Pull request closed without merging**